### ISSUE TITLE
Different result on converting keras to graph model

### ISSUE THREAD

**Issue by mmmahdiii** [tensorflow/tfjs#5686]
Hi
I converted a tensorflow keras model from Python to JavaScript graph model and the program works fine, but the model is much less accurate than the Python model.
I checked all the inputs and the program, all the values ​​were correct.
I tried all the conversion parameters but it did not work.

**Comment by vladmandic** [comment]
which backend? if its `wasm`, not surprised, see #5641  
did you quantize weights? if yes, you might be running into values clipping

**Comment by mmmahdiii** [comment]
@vladmandic 
thanks for your replying.

> which backend?

I have  the same problems with any backends, **WebGL, WASM or CPU**

> did you quantize weights? 

I tried this switches quantize_float16, quantize_uint8, quantize_uint16, old way quantization_bytes{1,2}  **and no quantize** 

> is your model using very large values?

My model use float32 


### and my Conversion  is:

2021-10-02 07:56:55.577985: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-10-02 07:56:56.330169: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1510] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 2153 MB memory:  -> device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5
**WARNING**:tensorflow:Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.
2021-10-02 07:57:15.603301: W tensorflow/python/util/util.cc:348] Sets are not currently considered sequences, but this may change in the future, so consider avoiding using them.
C:\Users\admin\.conda\envs\tfgpu\lib\site-packages\keras\utils\generic_utils.py:494: **CustomMaskWarning**: Custom mask layers require a config and must override get_config. When loading, the custom mask layer must be passed to the custom_objects argument.
  **warnings**.warn('Custom mask layers require a config and must override '
2021-10-02 07:57:49.538891: I tensorflow/core/grappler/devices.cc:66] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 1
2021-10-02 07:57:49.539553: I tensorflow/core/grappler/clusters/single_machine.cc:357] Starting new session
2021-10-02 07:57:49.545380: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1510] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 2153 MB memory:  -> device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5
2021-10-02 07:57:49.679314: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:1137] Optimization results for grappler item: graph_to_optimize
  function_optimizer: Graph size after: 2969 nodes (2202), 4589 edges (3821), time = 78.26ms.
  function_optimizer: function_optimizer did nothing. time = 1.917ms.

2021-10-02 07:57:52.382050: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:1137] Optimization results for grappler item: graph_to_optimize
  model_pruner: Graph size after: 2201 nodes (-765), 3821 edges (-765), time = 41.385ms.
  constant_folding: Graph size after: 1441 nodes (-760), 3061 edges (-760), time = 96.044ms.
  arithmetic_optimizer: Graph size after: 1459 nodes (18), 3120 edges (59), time = 24.767ms.
  dependency_optimizer: Graph size after: 1407 nodes (-52), 1505 edges (-1615), time = 128.634ms.
  model_pruner: Graph size after: 1407 nodes (0), 1505 edges (0), time = 9.362ms.
  constant_folding: Graph size after: 1407 nodes (0), 1505 edges (0), time = 28.538ms.
  arithmetic_optimizer: Graph size after: 1407 nodes (0), 1505 edges (0), time = 21.171ms.
  dependency_optimizer: Graph size after: 1407 nodes (0), 1505 edges (0), time = 12.314ms.
  debug_stripper: debug_stripper did nothing. time = 1.452ms.
  model_pruner: Graph size after: 1407 nodes (0), 1505 edges (0), time = 7.574ms.
  constant_folding: Graph size after: 1407 nodes (0), 1505 edges (0), time = 28.541ms.
  arithmetic_optimizer: Graph size after: 1407 nodes (0), 1505 edges (0), time = 22.465ms.
  dependency_optimizer: Graph size after: 1407 nodes (0), 1505 edges (0), time = 12.542ms.
  model_pruner: Graph size after: 1407 nodes (0), 1505 edges (0), time = 10.61ms.
  constant_folding: Graph size after: 1407 nodes (0), 1505 edges (0), time = 27.608ms.
  arithmetic_optimizer: Graph size after: 1407 nodes (0), 1505 edges (0), time = 22.503ms.
  dependency_optimizer: Graph size after: 1407 nodes (0), 1505 edges (0), time = 12.364ms.
  debug_stripper: debug_stripper did nothing. time = 1.486ms.

2021-10-02 07:58:01.840274: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:1137] Optimization results for grappler item: graph_to_optimize
  remapper: Graph size after: 1334 nodes (-221), 830 edges (-221), time = 33.389ms.
  constant_folding: Graph size after: 732 nodes (-602), 830 edges (0), time = 38.634ms.
  arithmetic_optimizer: Graph size after: 732 nodes (0), 830 edges (0), time = 15.25ms.
  dependency_optimizer: Graph size after: 732 nodes (0), 830 edges (0), time = 8.385ms.
  remapper: Graph size after: 732 nodes (0), 830 edges (0), time = 5.251ms.
  constant_folding: Graph size after: 732 nodes (0), 830 edges (0), time = 18.008ms.
  arithmetic_optimizer: Graph size after: 732 nodes (0), 830 edges (0), time = 14.307ms.
  dependency_optimizer: Graph size after: 732 nodes (0), 830 edges (0), time = 10.187ms.

Writing weight file ./model.json...

**Comment by vladmandic** [comment]
> Compiled the loaded model, but the compiled metrics have yet to be built

how was this keras model created?  
id try exporting kernas model as a tf frozen model and converting that to make sure everything is there

> CustomMaskWarning: Custom mask layers require a config and must override get_config

has this been done?

**Comment by mmmahdiii** [comment]
@vladmandic 

> how was this keras model created?

It's a keras model with tensorflow backend.

> id try exporting kernas model as a tf frozen model and converting that to make sure everything is there

I try tf Frozenmodel but it makes too much less accurate.


>> **CustomMaskWarning**: Custom mask layers require a config and must override get_config

>has this been done?

I don't have this warning in python, and I couldn't solve this warning in converting.
- how can I solve **CustomMaskWarning** ?
- how important this warning?

### Have you another idea to try?

**Comment by vladmandic** [comment]
> I try tf Frozenmodel but it makes too much less accurate.

If your model when frozen produces less accurate results in python, then that is the issue - if it cannot be properly frozen, you'll have the same issues when converted to TFJS graph model. So its not a TFJS issue, but a *"how to freeze a Keras model"* issue.

> Compiled the loaded model, but the compiled metrics have yet to be built

To me this looks like a big issue.

**Comment by mmmahdiii** [comment]
> If your model when frozen produces less accurate results in python
in python everything is ok. but frozen model that converted to to TFJS  produces less accurate results.
I'm sure it's the converting issue.
